Python/dynload_aix.c

```diff
@@ -144,12 +144,13 @@ aix_loaderror(const char *pathname)
         ERRBUF_APPEND(message[i]);
         ERRBUF_APPEND("\n");
     }
-    errbuf[strlen(errbuf)-1] = '\0';            /* trim off last newline */
-    pathname_ob = PyUnicode_FromString(pathname);
-    errbuf_ob = PyUnicode_FromString(errbuf);
-    PyErr_SetImportError(errbuf_ob, NULL, pathname);
-    Py_DECREF(pathname_ob);
-    Py_DECREF(errbuf_ob);
+    /* Subtract 1 from the length to trim off trailing newline */
+    errbuf_ob = PyUnicode_DecodeFSDefaultAndSize(errbuf, strlen(errbuf)-1);
+    pathname_ob = PyUnicode_DecodeFSDefault(pathname);
```

Comment:
`PyErr_SetImportError()` can fail if `errbuf_ob` or `pathname_ob` are NULL.

```suggestion
    errbuf_ob = PyUnicode_DecodeFSDefaultAndSize(errbuf, strlen(errbuf)-1);
    if (errbuf_ob == NULL) {
        return;
    }
    pathname_ob = PyUnicode_DecodeFSDefault(pathname);
    if (pathname_ob == NULL) {
        Py_DECREF(errbuf_ob);
        return;
    }
```

Reply:
Since we now check for NULL prior to calling PyErr_SetImportError, I remove the NULL checks I had previously added before calling `Py_DECREF` on the two objects.